### PR TITLE
Avoid kubelet API in Conformance tests

### LINKED ISSUE #81052: Conformance tests should not require access to kubelet API
**What happened**: 
In reviewing  a test for promotion to conformance I noticed the kubelet API was being accessed via the API server's node proxy endpoint (ref: https://github.com/kubernetes/kubernetes/pull/80731#discussion_r310836102).  This seemed off, so I reviewed [our conformance test requirements doc](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements).  It doesn't explicitly call out kubelet API access as allowed or disallowed.

[I asked #k8s-conformance if we had an opinion on this](https://kubernetes.slack.com/archives/C78F00H99/p1565050383095300).  @timothysc and @liggitt agreed kubelet API access seems like a bad idea.

I'm now surveying what we have that _does_ assume kubelet API access.  It's definitely used by parts of e2e framework and some Conformance tests. I'm trying to verify how critical its use is.  It appears that much of it is related to debugging or verifying whether kubelets and the control plane agree on who is running what.

**What you expected to happen**: 
Find a blanket policy that bans kubelet API access for Conformance tests, whether directly, or indirectly through the API server's node proxy endpoint, at any time.

Alternatively, find a policy that says kubelet API access may be allowed, but only for debugging or failure purposes (must not be on the critical path to make the test pass)

Alternatively, find a policy that says kubelet API access is allowed, but only indirectly through the API server, and evaluated against the same constraints we use for control plane access (GA, non-optional)

If we want to say kubelet API access is banned, we can use this as a tracking issue to demote whichever tests rely on it, and come up with a plan to verify the same behavior without kubelet API access.

/area conformance
/sig architecture
/priority important-soon
/milestone v1.16
/assign

### PR DESCRIPTION
**Pull request by spiffxp**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removes or avoids calls to the kubelet API in Conformance tests as
identified in https://github.com/kubernetes/kubernetes/issues/81052#issuecomment-561353611

**Which issue(s) this PR fixes**:

Fixes #81052

**Special notes for your reviewer**:
See individual commits for details on why/how the removal

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig architecture
/area conformance

### CODE REVIEW

**Comment by spiffxp** [comment]
/uncc @dashpole @k82cn 
/hold
for comment
/cc @johnbelamaric 
conformance
/cc @yujuhong 
for the change to test/e2e/common/pods.go
/cc @Huang-Wei 
for the change to test/e2e/scheduling/predicates.go

**Comment by Huang-Wei** [comment]
Scheduler e2e test seems to only use the kubelet node proxy API debugging the Pods on the specific node, so the changes LGTM.

/approve
on scheduling side.

File: test/e2e/common/pods.go [hunk 1]
```diff
@@ -301,28 +300,6 @@ var _ = framework.KubeDescribe("Pods", func() {
 		err = podClient.Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(30))
 		framework.ExpectNoError(err, "failed to delete pod")
 
-		ginkgo.By("verifying the kubelet observed the termination notice")
```
Comment (yujuhong) [inline test/e2e/common/pods.go:304]: I think we have the code here mostly for debugging, as it'd help narrow down the cause. 
It should be safe to remove and there are other tests in node e2e that will probably provide the same information.

**Pull request merged**